pkgpanda/__init__.py

```diff
@@ -42,14 +41,27 @@
 version_regex = "^[a-zA-Z0-9@_+:.]+$"
 username_regex = "^dcos_[a-z0-9_]+$"
 
+dcos_sysctl_config = '/etc/sysctl.d/dcos.conf'
+
+
+def preserve_sysctl(sysctl_config_file, preserved_sysctl_map):
```

Comment:
"preserve" I think is the wrong word here. It's much more a "make durable".

We also need to do a proper "swap" of the old file for the new file. Currently this overwrites the old values _before_ we get into the details of swapping packages, meaning that before we reach the "atomic" part of an upgrade, we might have made it so we can't roll back properly.